src/lib/storage/dictionary_segment.hpp

```diff
@@ -22,9 +23,17 @@ class DictionarySegment : public BaseDictionarySegment {
   explicit DictionarySegment(const std::shared_ptr<const pmr_vector<T>>& dictionary,
                              const std::shared_ptr<const BaseCompressedVector>& attribute_vector);
 
+  explicit DictionarySegment(const std::shared_ptr<const std::span<const T>>& dictionary,
+                             const std::shared_ptr<const BaseCompressedVector>& attribute_vector);
+
+  explicit DictionarySegment(const uint32_t* start_address);
```

Comment:
I do not understand too much of explicit constructors, but should this maybe also be a shared ptr?

Reply:
that would be a shared_pointer on a pointer and especially a shared_pointer on one memory address whose existence and validity depends on the existence of a map. We need to handle memory management  for mmap-based structures somewhere (and think about how exactly to implement it), but for now I think this is fine.